### DISCUSSION TITLE
NEAR Lake Framework 0.8.0 beta (High-level) release Feedback wanted

### DISCUSSION THREAD

**Discussion by khorolets** [near/near-lake-framework-rs#84] [Announcements]
We've released a beta version of the NEAR Lake Framework 0.8.0 (High-level update).

The [post on NEAR.org](https://near.org#/near/widget/PostPage?accountId=khorolets.near&blockHeight=93659695) where we asked for your feedback. This discussion post is the best place to give feedback, ask questions and raise concerns. We are here to listen and help you!

**Comment by khorolets** [comment]
It has been a while since I posted this. I haven't received any feedback, proposals, or questions regarding the high-level update for the NEAR Lake Framework.

I had some doubts about the necessity of this update. While such simplification might make sense for the JavaScript framework version, it makes little sense for the Rust world. 

My initial intention was to keep both framework versions close to each other, and now it doesn't make sense either.

This is the last callout to anyone who believes the high-level update makes sense and is worth finishing the upgrade and migrating to a new era of NEAR Lake Framework. Please let me know if you are here. Otherwise, I will consider rejecting the idea of a high-level update for 0.8.0 and will stick to the approach we used to have all this time (0.7.x).

Thank you.

**Comment by frol** [comment]
It is nice to have both low-level and high-level APIs, so given that high-level API was already well-designed and low-level API is still available (so no breaking changes to the existing code), I’d love to see high-level API released.

Are there any unresolved questions there?

**Reply by khorolets** [reply]
> Are there any unresolved questions there?

It's mostly capacity issues. Although there are a few things here and there to adjust, it's mostly about time investments in documenting. My initial approach to the documentation of the high-level update was "before-after," but then I realized it was the wrong approach. 
The proper way to document is for those who start using the library in mind.

**Comment by ewiner** [comment]
I agree, if you haven't seen people clambering to use 0.8, there's no reason to spend time pushing forward with it and maintaining the new API indefinitely.  Are you aware of anyone using the 0.8 beta?

**Comment by ChaoticTempest** [comment]
Currently, [chain-signatures](https://github.com/near/mpc/tree/develop/chain-signatures) uses the high level API, and I think it's great to have. Haven't used the more extensive features, but the ease of usage lends to not needing to know the details of the lake framework. It lets us focus on what matters so I would vote for maintaining the high level API if it's not too much of a maintenance churn. However, we're using the most basic of features here (just spinning up the indexer with a Context object and listening for a specific function) so no huge priority from us to have more features.
